go/v1beta1/storage/storagefactory.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/grafeas/grafeas/go/config"
+	"github.com/grafeas/grafeas/go/v1beta1/api"
+	"github.com/grafeas/grafeas/go/v1beta1/project"
+)
+
+// use type aliasing to get multiple inheritance even though both interfaces are called Storage
+type Ps = project.Storage
+type Gs = grafeas.Storage
+
+// unified interface that basically gives us a single interface called Storage that implements
+// both anonymous interfaces.
+type Storage struct {
+	Ps
+	Gs
+}
+
+var registeredStorageTypeProviders = map[string]func(storageType string, storageConfig *interface{}) (*Storage, error){}
+
+// RegisterStorageTypeProvider registers a new provider to create a specific type of Storage
+func RegisterStorageTypeProvider(storageType string, provider func(storageType string, storageConfig *interface{}) (*Storage, error)) error {
+	if _, present := registeredStorageTypeProviders[storageType]; !present {
+		registeredStorageTypeProviders[storageType] = provider
+		return nil
+	} else {
+		return errors.New(fmt.Sprintf("Storage provider %s already exists", storageType))
+	}
+}
+
+// CreateStorageOfType will create an instance of Storage by name or an error if that type is unsupported.
+func CreateStorageOfType(storageType string, storageConfig *interface{}) (*Storage, error) {
+	if provider, present := registeredStorageTypeProviders[storageType]; present {
+		return provider(storageType, storageConfig)
+	} else {
+		return nil, errors.New(fmt.Sprintf("Unsupported storage type %s", storageType))
+	}
+
+}
+
+// MemstoreStorageTypeProvider returns a memstore storage instance
```

Comment:
nit: lower case 'm' in MemsoterStorageTypeProvider

Reply:
ditto for the functions below

Reply:
Amended.